advanced/cli/index.md

```diff
@@ -55,7 +55,41 @@ time it is run.
 
 ## Usage
 
-Run `tugboat help` to see a list of commands that can be executed
+Run `tugboat help` to see a list of commands that can be executed. Run
+`tugboat <command> -help` to see detailed documentation of a specific commands.
+For example, `tugboat api -help` will explain how the `api` command works.
+
+## Commands
+
+Here are the commands available to the Tugboat CLI. These commands are found by
+calling `tugboat help` from the command line.
+
+```
+api <type> <action> [args...]      Execute a Tugboat API call directly
+cancel <id>                        Cancel a preview that is currently building
+clone <id> [args...]               Clone a Tugboat Preview
+config <repo> [command] [args...]  Configure a Tugboat Repository
+create <type> [name] [args...]     Create a Tugboat resource
+delete|rm <id> [args...]           Delete a Tugboat resource
+find <id>                          Find a Tugboat resource
+grant <id> [args...]               Grant access to a Tugboat resource
+history <id> [args...]             History of Tugboat resource
+list|ls <id> [args...]             List a Tugboat resource
+log [options] <id> [args...]       View the log for a Tugboat resource
+ping <id>                          Validate the authentication of a Tugboat repository
+rebuild <id> [args...]             Rebuild a Tugboat Preview
+refresh <id> [args...]             Refresh a Tugboat Preview
+rekey <id>                         Rekey a Tugboat Preview
+reset <id>                         Reset a Tugboat Preview or Service
+revoke <id> [args...]              Revoke access from a Tugboat resource
+shell <id> [args...]               Start a shell session in a Tugboat resource
+start <id>                         Start a Tugboat Preview
+stop <id>                          Stop a Tugboat Preview
+suspend <id>                       Suspend a Tugboat Preview
+update <id> [args...]              Update a Tugboat resource
+validate <id|filename> [ref]       Validate a tugboat config.yml
+help [cmd]                         display help for [cmd]
+```
```

Comment:
I wonder if there would be a way to auto-populate this from the latest tugboat command. I worry this will get out of date otherwise.

Reply:
This is why we originally kept this page brief. The intention was to provide a few examples, but point to the in-line help in the CLI for more details because that would always be kept up to date.

I think I'd like to keep it that way. Looping in @dacharyc for her thoughts

Reply:
I think there are arguments to be made both ways. On the one hand, viewing reference materials inside the command line can be a PITA while also working in the command line, so having a reference page on a site somewhere while working can be helpful. On the other hand, if the options change frequently, getting an outdated reference is frustrating for a user. (If I were a user working in the CLI, and had to query the help more than once to look something up, I'd probably just copy the response to a doc while I'm working so I have the details handy.)

I guess I'd ask how often this changes, and if it's reasonable to do a periodic audit - like monthly cross-referencing in-line help in the CLI vs. external reference page - or adding a step to workflow when updating tugboat command to update the external doc in step. Or if it's possible to set up a script to update the external doc when changes are made to tugboat command to automate doc updates. If the answer is that it's not feasible to keep both updated, I'd say don't publish the details in an external doc; instruct users to query the in-line help in the CLI for options.

Reply:
I thought it might be handy to at least show the overall set of CLI commands in the documentation so users get a better sense of why they might want to download the tool and what they could do with it. 

One thing Adam asked for was for more CLI examples, but I'm not able to provide that because I don't use the CLI tool often enough. 

So I'm okay dropping the list of CLI commands if it creates a lot of additional work in issuing documentation updates.

Reply:
If that's the case, I'd say it makes sense to offer the info up, with the caveat that the list may not be complete, and to query the in-line help in the CLI for the most recent set of commands.

Reply:
Okay. I added a warning that the commands may be deprecated. I think this might be good to go now?

Reply:
I think the warning callout is a bit overkill, and draws too much attention to the fact that we probably don't update our documentation very frequently.